test/grouping.jl

```diff
@@ -4378,4 +4378,45 @@ end
     @test_throws ArgumentError combine(gdf, :a => (x -> [Dict('x' => 1)]) => AsTable)
 end
 
+@testset "sorting API" begin
+    Random.seed!(1234)
+    df = DataFrame(a=rand(-10:10, 100), b=rand(-10:10, 100), c=1:100)
+    for col in (:a, "a", 1, :b, "b", 2, :c, "c", 3) 
+        gdf = groupby(df, col, sort=true)
```

Comment:
Test that the result isn't sorted when `sort=false`, so that if the algorithm changes in the future we notice that these tests need to be adapted?

Reply:
OK - added an appropriate test for `sort=false` both for single column `groupby` and multi-column `groupby`.